frontend/src/pages/RunDetails.tsx

```diff
@@ -677,12 +677,11 @@ class RunDetails extends Page<RunDetailsInternalProps, RunDetailsState> {
         !jsonWorkflow.status.nodes &&
         jsonWorkflow.status.compressedNodes
       ) {
-        jsonWorkflow.status.nodes = await decodeCompressedNodes(
-          jsonWorkflow.status.compressedNodes,
-        );
+        jsonWorkflow.status.nodes = decodeCompressedNodes(jsonWorkflow.status.compressedNodes);
```

Comment:
@Bobgy I am not an expert with React testing but I can't understand why I needed to remove `await` here for tests to pass! If I keep `await` code fails and the page is empty!

Reply:
Hey, I think there are two problems showing together:
1. the code should have `await` there, but there's sth wrong in the test to make it wait for the gunzip decompress properly
2. when you remove the `await`, the test no longer needs to wait gunzip decompress, so it shows sth in the snapshot, but the snapshot doesn't show any **real** content in there either. That's a problem with our current way for testing this component.

Would you mind waiting a few days to let me refactor this one node graph test case first to solve problem 2? You can mimic my refactored test case to properly test it and figure out what went wrong to problem 1.

Reply:
Also, these are not specific to react, they are javascript testing problems.

To explain problem 1 a bit more, we use `await flushPromises()` in tests to wait for mocked promises that return immediately, but you didn't mock the zlib library here, it's not clear when it will run the compress callback handler.

For this specific case, I don't see why you cannot use the `gunzipSync` method instead: https://nodejs.org/api/zlib.html#zlib_zlib_gunzipsync_buffer_options
It will avoid your problem.

Reply:
Thanks @Bobgy for your quick response. Actually I can mock also the `decodeCompressedNodes` so we escape the problem with `flushPromises`.

I used async `gunzip` to avoid freezing UI while decoding compressed nodes. Usually, server compress the nodes property when the pipeline is too big, so it may takes some time to decode the base64 and gunzip it.

I don't know if mocking `decodeCompressedNodes` will fix Problem 2, we will see.